app/models/miq_request_workflow.rb

```diff
@@ -344,12 +344,19 @@ def rails_logger(_name, _start)
     # Rails.logger.warn("#{name} #{start.zero? ? 'start' : 'end'}")
   end
 
+  def self.handle_deprecation(text)
+    deprecated_warn = "method: parse_ws_string, arg Type => String"
+    solution = "arg should be a hash"
+    MiqAeMethodService::Deprecation.deprecation_warning(deprecated_warn, solution) if text.kind_of?(String)
+  end
+
   def parse_ws_string(text_input, options = {})
     self.class.parse_ws_string(text_input, options)
   end
 
   def self.parse_ws_string(text_input, options = {})
     return parse_request_parameter_hash(text_input, options) if text_input.kind_of?(Hash)
+    handle_deprecation(text_input)
```

Comment:
Move the deprecation logic down one line and then you will not need to check for `kind_of?(String)`